repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -178,19 +184,27 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                         }
 
                         for (String asset : assets) {
-                            addPersonaMetadataFilterClauses(asset, allowClauseList);
+                            terms.add(asset);
+                            allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset + "/*")));
                         }
 
-                        addPersonaMetadataFilterConnectionClause(connectionQName, allowClauseList);
+                        terms.add(connectionQName);
 
                     } else if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_GLOSSARY)) {
                         for (String glossaryQName : assets) {
-                            addPersonaGlossaryFilterClauses(glossaryQName, allowClauseList);
+                            terms.add(glossaryQName);
+                            allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, "*@" + glossaryQName)));
                         }
                     }
                 }
             }
         }
+
+        if (terms.size() > assetsMaxLimit) {
```

Comment:
Just an optimization thought: I'm not sure about what `terms.size()` can balloon up to, but should we consider throwing exception sooner than end of preparation for `terms`, maybe as an `if` clause somewhere in the loop throw exception as soon as we reach `assetsMaxLimit`.

Reply:
Addressed

Reply:
terms.size() + assets.size() + 1 for Metadata policies where along with assets we add 1 more clause for connection qualifiedName hence +1

For Glossary policies, it is only the assets count.